src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -720,9 +718,6 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_do_accumulate(const void *origin_addr,
         basic_count = target_bytes / dt_size;
         MPIR_Assert(target_bytes % dt_size == 0);
 
-        /* Ensure completion of outstanding AMs for atomicity. */
-        MPIDIG_wait_am_acc(win, target_rank);
```

Comment:
```
MPIDIG_wait_am_acc is insufficient to guarantee the completion of active
message atomics since there is no way of accounting for remote
initiating active messages that is targeting the process.
```
My understanding is that atomicity is maintained for operations issued to the same target memory region from the same origin process. The origin should always be able to determine if active messages have been used that need to be completed, or am I missing something?

Reply:
No, atomicity matters for operations to the same target memory region from **different** origin processes as well. Let's say the NIC hardware is updating a memory, and the CPU tries to update the same memory at the same time. Both operation may be originated from different process, but they are being updated simultaneously, potentially.

Reply:
From MPI 3.1 section 11.7.2
```
Ordering only applies to operations originating at the same
origin that access overlapping target memory regions. MPI does not provide any guarantees
for accesses or updates from different origin processes to overlapping target memory regions.
```

Reply:
Not ordering, but atomicity. The fact that having CPU and hardware updating the same memory potentially can leave corrupted data.

Reply:
MPI makes no guarantee about atomicity when the operations originate at different origin processes. But if they originate from the same origin, then atomicity **and ordering** must be maintained. If some other origin process is potentially corrupting the date in the same memory region, then that is an application problem.

Reply:
Hui is correct. Atomicity is for different origin processes, only ordering is for the same origin processes.

I actually just found that my previous thought for `MPIDIG_wait_am_acc` is incorrect. It may happen that two accumulates issued by the **same origin** and access to **different memory location** on the target process. We need ensure ordering for such case, but no atomicity guarantee.

So we need revert commit `4ec65a0631abf22bbaf51dedbf911fcb095a111d`.

Reply:
OK, my mistake on the atomicity part.